### PR TITLE
New plots to analyse Nitrogen abundance

### PR DESCRIPTION
**Pull request by correac**
New plots for log10(C/O) vs 12+log10(O/H) and log10(N/O) vs 12+log10(O/H). The goal of these relations is to test the primary and secondary production channels of Nitrogen nucleosynthesis.

### CODE REVIEW

File: colibre/registration.py [hunk 1]
```diff
@@ -845,10 +845,157 @@ def register_star_Mg_and_O_to_Fe(self, catalogue, aperture_sizes):
...
+    for aperture_size in aperture_sizes:
+
+        # Fetch N over O times gas mass computed in apertures. The
+        # mass ratio between N and O has already been accounted for.
+        log_N_over_O_times_gas_mass = catalogue.get_quantity(
+            f"lin_element_ratios_times_masses.lin_N_over_O_times_gas_mass_{aperture_size}_kpc"
+        )
+        # Fetch gas mass in apertures
+        gas_mass = catalogue.get_quantity(f"apertures.mass_gas_{aperture_size}_kpc")
+        # gas_cold_dense_mass = catalogue.get_quantity(
```
Comment (robjmcgibbon) [inline colibre/registration.py:859]: Please remove commented code

**Review by robjmcgibbon — COMMENTED** [review]
We need to wait until https://github.com/SWIFTSIM/velociraptor-comparison-data/pull/179 is merged so you can update the submodule link

**Comment by robjmcgibbon** [comment]
Could you please also upload some images of the new plots

File: colibre/registration.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (correac) [reply colibre/registration.py:859]: For this case I was testing the change in the abundance ratios. But the default quantity is cold dense gas to mostly account for ISM gas.

Comment (correac) [reply colibre/registration.py:859]: Yes! I'm cleaning now..

**Comment by MatthieuSchaller** [comment]
Is this still [WIP] ?

**Comment by correac** [comment]
@MatthieuSchaller  I'm checking consistency in everything now since I have four merge requests (SOAP, velociraptor-python, pipeline-configs and observational_dataset). Between today and tomorrow I will remade a few plots and recheck all changes to remove the WIP.

**Comment by MatthieuSchaller** [comment]
Any news on this? Still WIP?

**Comment by correac** [comment]
New plots for log10(C/O) vs 12+log10(O/H) and log10(N/O) vs 12+log10(O/H). The new figures are shown below. @MatthieuSchaller @robjmcgibbon I reviewed the changes and I am happy, this PR can be merged if there are no outstanding issues.
image

image

image

**Comment by correac** [comment]
I am adding four new plots to this PR. I hope that's ok! The new plots correspond to the stellar abundance plots with some variations: (1) [N/O] vs [O/H], (2) [N/O] vs [Fe/H], (3) [C/O] vs [O/H], (4) [C/O] vs [Fe/H]. Find the new figures below.
image
image
image
image

File: colibre/registration.py [hunk 2]
```diff
@@ -845,10 +845,205 @@ def register_star_Mg_and_O_to_Fe(self, catalogue, aperture_sizes):
...
+
+def register_carbon_to_oxygen(self, catalogue, aperture_sizes):
+    # Loop over aperture average-of-linear C/O-abundances
+    for aperture_size in aperture_sizes:
+
+        # Fetch C over O times gas mass computed in apertures. The
+        # mass ratio between N and O has already been accounted for.
+        # Note that here we are calling the diffuse quantities, for
+        # total quantities (diffuse + dust) call "lin_C_over_O_total_times_ .."
+        log_C_over_O_times_gas_mass = catalogue.get_quantity(
```
Comment (robjmcgibbon) [inline colibre/registration.py:937]: Is it possible to copy the loop [here](https://github.com/correac/pipeline-configs/blob/8452738affeb21bd79ebb48c254b90512c1a958a/colibre/registration.py#L1047) to register the diffuse and total quantities with a bit less code?

File: colibre/auto_plotter/metallicity.yml [hunk 3]
```diff
@@ -498,6 +498,299 @@ stellar_mass_gas_sf_metallicity_lom_allgals_50:
     - filename: GalaxyStellarMassGasMetallicity/FIREbox.hdf5
     - filename: GalaxyStellarMassGasMetallicity/Strom2022_OH.hdf5
 
+stellar_mass_gas_nitrogen_over_oxygen_lom_50_kpc:
+  type: "scatter"
+  legend_loc: "upper left"
+  selection_mask: "derived_quantities.has_cold_dense_gas_50_kpc"
+  comment: "All galaxies, LoM, Diffuse"
```
Comment (robjmcgibbon) [inline colibre/auto_plotter/metallicity.yml:505]: Should this be updated to reflect it's only galaxies with cold dense gas?

**Review by robjmcgibbon — COMMENTED** [review]
Looks pretty good to me, but I have a minor query about plot comments, and then a request to cut down some code

File: colibre/auto_plotter/metallicity.yml (hunk 3 above)
Comment (correac) [reply colibre/auto_plotter/metallicity.yml]: I updated all comments to describe galaxies with cold dense gas. Also, in metallicity.ylml I modified the captions of some figures from the stellar metallicity section that needed improving.

File: colibre/registration.py (hunk 2 above)
Comment (correac) [reply colibre/registration.py]: Yes! Good point. I included the loop in both the C/O and N/O registration functions.

**Comment by robjmcgibbon** [comment]
Perfect, all looks good to me! Are you happy to merge https://github.com/SWIFTSIM/velociraptor-comparison-data/pull/179? We need to update the submodule link before we merge this PR

**Comment by correac** [comment]
Yes! I am happy to merge https://github.com/SWIFTSIM/velociraptor-comparison-data/pull/179 . I reviewed it the other day and it looks good. Thank you Rob!!

**Pull request merged**